### PR TITLE
add tests for UI module

### PR DESCRIPTION
**Pull request by HubLot**
Here some tests for the UI module.
It will make sure the API and the entry point behave as they should.

By the way, I spotted a bug when using partial def file. When I implemented the function `check_def_topol_consistency`, I expected that all carbons in the top file are present in the def file (which isn't the case when using partial def file) hence buildH crashed due to a `keyerror`. I fixed it

### CODE REVIEW

File: buildh/init_dics.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def make_dic_atname2genericname(filename):
                 name, _, C, H = line.split()
                 dic[(C, H)] = name
     except:
-        raise ValueError("Can't read order parameter definition in "
+        raise ValueError("Something went wrong wen parsing the order parameter definition in "
```
Comment (patrickfuchs) [inline buildh/init_dics.py:40]: when

File: tests/test_dics.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ def test_make_dic_atname2genericname(inputs):
     # Test non existent file
     with pytest.raises(ValueError) as err:
         init_dics.make_dic_atname2genericname("none")
-    assert "Can't read order parameter" in str(err.value)
+    assert "Something went wrong wen parsing the order parameter definition in file" in str(err.value)
```
Comment (patrickfuchs) [inline tests/test_dics.py:59]: when

**Review by patrickfuchs — COMMENTED** [review]
Spotted a couple of typos

**Comment by patrickfuchs** [comment]
>Here some tests for the UI module.
>It will make sure the API and the entry point behave as they should.

Great!

>By the way, I spotted a bug when using partial def file. When I implemented the function check_def_topol_consistency, I  expected that all carbons in the top file are present in the def file (which isn't the case when using partial def file) hence buildH crashed due to a keyerror. I fixed it

This bug was occuring when using **buildH** as a module, right?

**Comment by HubLot** [comment]
no, all  the time

**Comment by patrickfuchs** [comment]
> no, all the time

Weird, I used a partial def file not long ago and didn't get that error. When do you think this bug was introduced?

**Comment by HubLot** [comment]
hum, can you try the master before merging then ?
It was added during this commit I guess : 311bd2840d1db9305b62ccaa95a8d8be2acbc295

**Comment by patrickfuchs** [comment]
> hum, can you try the master before merging then ?

Oops indeed you're right, I just obtained a `KeyError: 'C1'` !!!

In fact, I probably used a complete def file. I probably mixed things, sorry 😆

So good you fixed this! Shall I go for the merge?

**Comment by HubLot** [comment]
good, yes!

**Pull request merged**